### DISCUSSION TITLE
Marpit Markdown and auto-scaling features: how to use it?

### DISCUSSION THREAD

**Discussion by aborruso** [marp-team/marp#186] [Q&A] [answered]
Hi,
I would like (if it's possible) to apply auto-scaling in Marpit Markdown.

How to set it in Front-matter?

In this way I think it's wrong, because I have two times `theme`:

```
---
marp: true
paginate: true
theme: uncover
theme: enable-all-auto-scaling
auto-scaling: true

---
```

Thank you

**Comment by yhatt** [comment, marked as answer]
It is a specific feature for [Marp Core's extended CSS theming system](https://github.com/marp-team/marp-core/tree/main/themes#metadata-for-additional-features). Both of [Marp Markdown for Marp Core](https://github.com/marp-team/marp-core#marp-markdown) and [Marpit Markdown for Marpit framework](https://marpit.marp.app/markdown) have not supported defining auto scaling in the front-matter. `auto-scaling` directive for Markdown front-matter has _ever never provided_ in any Marp ecosystem tools. (Even though, [many people are defined it in the front-matter WRONGLY](https://github.com/search?l=Markdown&o=desc&q=auto-scaling+marp&s=indexed&type=Code))

If enabled auto scaling, some side effects for getting auto scaling will be affected to DOM. The theme CSS author has to turn on auto-scaling at a risk, and handle side effects correctly. Thus, _the Markdown author cannot change this setting, to prevent breaking the design provided by theme CSS._

If you want to use auto scaling, you have to make a custom theme with enabled auto scaling through the metadata:

```css
/* custom-theme.css */

/**
 * @theme your-custom-theme
 * @auto-scaling true
 */

section {
  ...
}
```

```markdown
---
theme: your-custom-theme
---

# <!--fit--> Enabled auto scaling
```
